src/auto/seat.rs

```diff
@@ -70,10 +78,24 @@ impl Seat {
         }
     }
 
-    //#[cfg(any(feature = "v3_20", feature = "dox"))]
-    //pub fn grab<'a, 'b, 'c, P: IsA<Window>, Q: Into<Option<&'a Cursor>>, R: Into<Option<&'b Event>>, S: Into<Option<&'c /*Unimplemented*/SeatGrabPrepareFunc>>, T: Into<Option</*Unimplemented*/Fundamental: Pointer>>>(&self, window: &P, capabilities: SeatCapabilities, owner_events: bool, cursor: Q, event: R, prepare_func: S, prepare_func_data: T) -> GrabStatus {
-    //    unsafe { TODO: call ffi::gdk_seat_grab() }
-    //}
+    #[cfg(any(feature = "v3_20", feature = "dox"))]
+    pub fn grab<'a, 'b, P: IsA<Window>, Q: Into<Option<&'a Cursor>>, R: Into<Option<&'b Event>>, S: FnMut(&Seat, &Window), T: Into<Option<S>>>(&self, window: &P, capabilities: SeatCapabilities, owner_events: bool, cursor: Q, event: R, prepare_func: T) -> GrabStatus {
+        let cursor = cursor.into();
+        let event = event.into();
+        let prepare_func = prepare_func.into();
+        let prepare_func_data: Option<S> = prepare_func.into();
+        unsafe extern "C" fn prepare_func_func<'a, 'b, P: IsA<Window>, Q: Into<Option<&'a Cursor>>, R: Into<Option<&'b Event>>, S: FnMut(&Seat, &Window)>(seat: *mut ffi::GdkSeat, window: *mut ffi::GdkWindow, user_data: glib_ffi::gpointer) {
+            let seat = from_glib_borrow(seat);
+            let window = from_glib_borrow(window);
+            let callback: *mut S = user_data as *const _ as usize as *mut S;
```

Comment:
These functions here are actually broken the same way as elsewhere. We need to regenerate once gir is fixed @GuillaumeGomez